### PR TITLE
Disable HttpListener WebSocket test on Windows 7

### LINKED ISSUE #18906: Test: System.Net.Tests.HttpListenerRequestTests/IsWebSocketRequest_GetProperty_ReturnsExpected failed with "Xunit.Sdk.EqualException"
Opened on behalf of @Jiayili1

The test `System.Net.Tests.HttpListenerRequestTests/IsWebSocketRequest_GetProperty_ReturnsExpected(webSocketString: \"Connection: Upgrade\\r\\nUpgrade: websocket\", expected: True)` has failed.

Assert.Equal() Failure\r
    Expected: True\r
    Actual:   False

        Stack Trace:

           at System.Net.Tests.HttpListenerRequestTests.<>c__DisplayClass12_0.<IsWebSocketRequest_GetProperty_ReturnsExpected>b__0(Socket _, HttpListenerRequest request)
       at System.Net.Tests.HttpListenerRequestTests.<GetRequest>d__29.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Net.Tests.HttpListenerRequestTests.<IsWebSocketRequest_GetProperty_ReturnsExpected>d__12.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Build : Master - 20170425.01 (Core Tests)
Failing configurations:
- Windows.7.Amd64-x64
  - Debug
  - Release
- Windows.7.Amd64-x86
  - Debug
  - Release

Detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170425.01/workItem/System.Net.HttpListener.Tests/analysis/xunit/System.Net.Tests.HttpListenerRequestTests~2FIsWebSocketRequest_GetProperty_ReturnsExpected(webSocketString:%20%5C%22Connection:%20Upgrade%5C%5Cr%5C%5CnUpgrade:%20websocket%5C%22,%20expected:%20True)

### PR DESCRIPTION
**Pull request by davidsh**
HttpListener on Windows platforms requires "websocket.dll" native
component in order for WebSocket upgrade to work. And Windows platforms
before Windows 8 did not have that component.

Fixes #18906.

### CODE REVIEW

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 1]
```diff
@@ -195,6 +194,12 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
         [InlineData("Unknown-Header: Test", false)]
         public async Task IsWebSocketRequest_GetProperty_ReturnsExpected(string webSocketString, bool expected)
         {
+            // Skip on Windows 7 or UAP platforms. WebSocket support is not present.
+            if (PlatformDetection.IsWindows7 || !PlatformDetection.IsNotOneCoreUAP)
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: Don't we have a property somewhere that we could use from a ConditionalTheory?  I thought we had one we used in the WebSockets.Client tests?

Comment (davidsh) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: >Don't we have a property somewhere that we could use from a ConditionalTheory? I thought we had one we used in the WebSockets.Client tests?

I didn't see anything obvious that combines the "Windows 7 or UAP" check. So, just putting it into the test itself seemed like a simple solution.  Creating a new static property just for that and then making it into a ConditionTheory seemed like too much overhead.

Feel free to improve on next PR.

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 2]
```diff
@@ -186,7 +186,6 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
             });
         }
 
-        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
```
Comment (hughbe) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: I might be blind but aren't you removing a conditionaltheory without adding a theory? This test will never run

Comment (davidsh) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: Wow.  Good catch.  Yes, this needs a Theory attribute. We'll need to submit another PR to fix that.

Comment (hughbe) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: I'll handle it
